text/MichałZwonek/opis.tex

```diff
@@ -0,0 +1,173 @@
+\documentclass[12pt]{article}
+\usepackage[T1]{fontenc}
+\usepackage[utf8]{inputenc}
+\usepackage{amsmath}
+\usepackage{amsthm}
+\usepackage{hyperref}
+\usepackage{cleveref}
+\usepackage{fullpage}
+
+\setlength{\marginparwidth}{2cm}
+
+\theoremstyle{definition}
+\newtheorem*{theorem}{Twierdzenie}
+\newtheorem{lemma}{Lemat}
+\crefname{lemma}{}{}
+\newtheorem{definition}{Definicja}
+
+\title{Blum et al., Liniowa aproksymacja najkrótszego wspólnego nadsłowa, algorytm 3-przybliżony}
+\author{Michał Zwonek}
+\date{June 2020}
+
+
+\begin{document}
+
+\maketitle
+
+\section*{Wprowadzenie}
+
+Mając na wejściu słowa $S = \{s_1,...,s_m\}$, takie, że żadne z nich nie zawiera żadnego innego w całości, możemy poszukać wspólnego nadsłowo $s$ dla danego zbioru $S$.
+
+Najprostszym przykładem może być słowo $s=s_1s_2...s_m$. Jednakże, tak uzyskane nadsłowo będzie niepotrzebnie długie.
+
+Jako, że problem ten ma duże znaczenie praktyczne (w sekwencjonowaniu DNA) naturalnym jest próba znalezienia najkrótszego wspólnego nadsłowa.
+
+Oczywistym podejściem jest następujący algorytm zachłanny. Po kolei łącz słowa mające największą część wspólną nachodzącą na siebie określaną jako \textit{overlap}. Rób tak, aż do uzyskania 
+jednego słowa. Postawiona jest hipoteza, że takie podejście ma współczynnik aproksymacji równy 2, ale przed opublikowaniem pracy przez Blum et al. nie było wiadome, czy taki algorytm jest w ogóle liniową aproksymacją. Okazuje się, że ten algorytm (określać będziemy go poprzez \textbf{GREEDY}) ma współczynnik aproksymacji co najwyżej 4.
+
+Ponadto Blum et al. zaprezentowali alternatywny algorytm, nazywany \textbf{TGREEDY}, dla którego udowodnili, że jest on 3-aproksymacyjny. Sam problem znalezienia najkrótszego wspólnego nadsłowa jest NP trudny.
+
+W pierwszej sekcji przedstawimy podstawowe definicje i notacje, a w kolejnych algorytm \textbf{MGREEDY}, a w następynej sekcji algorytm 
```

Comment:
There is a description of greedy in the following part:

"Oczywistym podejściem jest następujący algorytm zachłanny. Po kolei łącz słowa mające największą część wspólną nachodzącą na siebie określaną jako _overlap_. Rób tak, aż do uzyskania 
jednego słowa. Postawiona jest hipoteza, że takie podejście ma współczynnik aproksymacji równy 2, ale przed opublikowaniem pracy przez Blum et al. nie było wiadome, czy taki algorytm jest w ogóle liniową aproksymacją. Okazuje się, że ten algorytm (określać będziemy go poprzez **GREEDY**) ma współczynnik aproksymacji co najwyżej 4."

However, I agree that the mention is easy to miss, so maybe I could rephrase the part you highlighted to something like this:

"W pierwszej sekcji przedstawimy podstawowe definicje i notacje, a w kolejnych algorytm **MGREEDY** będący pewną modyfikacją algorytmu **GREEDY**, taką dla którego analiza aproksymacji jest prostsza. W następnej sekcji 
przedstawimy algorytm **TGREEDY** wykorzystujący algorytm **MGREEDY** by uzyskać pożądany współczynnik aproksymacji. W ostatniej sekcji postaramy się udzielić odpowiedzi na pytanie, który z tych dwóch algorytmów jest lepszy?"

And maybe add something like this to the previous extract:

"[...]Okazuje się, że ten algorytm (określać będziemy go poprzez **GREEDY**) ma współczynnik aproksymacji co najwyżej 4, dowodu tego faktu nie będziemy przybliżać. Sam algorytm jest kluczowy, bo jest punktem startowym dla kolejnych dwóch algorytmów na których w tej pracy się skupimy."

What do you think, does that sound better?

Truth be told, the hardest part in writing the description of the original paper was actually compressing it to 3-4 pages, so that it was still readable, contained the main proof and did not exceed 4 pages. I might have misinterpreted this guideline as a hard upper bound, where it actually was more of an lower bound or a soft upper bound. That is my mistake and I think it might have resulted in poorer reader experience, sorry for that.

Reply:
We may leave it like that. I know that's an advice as an afterthought but in such cases do not hesitate to ask.

I'll try to clean this up when I'll be doing the pass to make the notations of all your notes and codes uniform (as that was taken as a soft bound instead, a bit over my expectations). I guess I'll make these three separate sections: greedy (known to be 4-opt), mgreedy (4-opt, but different), tgreedy (3-opt).

Reply:
Okay, I'll fix the other issues you've already pointed out and then I think I should have the updated PR ready before midnight.